kong/plugins/prometheus/exporter.lua

```diff
@@ -168,22 +168,30 @@ if kong_subsystem == "http" then
       return
     end
 
-    local service_name
+    local service_name, service_tags
     if message and message.service then
       service_name = message.service.name or message.service.host
+      service_tags = ((message.service.tags) and (type(message.service.tags) == "table"))
+                    and concat(message.service.tags, ",")
+                    or ""
     else
       -- do not record any stats if the service is not present
       return
     end
 
-    local route_name
+    local route_name, route_tags
     if message and message.route then
       route_name = message.route.name or message.route.id
+      route_tags = ((message.route.tags) and (type(message.route.tags) == "table"))
```

Comment:
I'm not sure if the sorting order of tags in Kong's DAO is stable or not i.e. can the order of tags change or not.
If it can change, I'd recommend sorting them because otherwise, it will lead to unnecessary duplication of metrics in the Prometheus store.

This is also additional table pollution probably. Wangchong will know best how to tackle that part.

Reply:
Right! I added sorting :smile: